pkg/controller/ingress-shim/sync.go

```diff
@@ -180,6 +181,22 @@ func (c *Controller) buildCertificates(ing *extv1beta1.Ingress, issuer v1alpha1.
 		if existingCrt != nil {
 			klog.Infof("Certificate %q for ingress %q already exists", tls.SecretName, ing.Name)
 
+			ownerRefs := existingCrt.GetOwnerReferences()
+			if len(ownerRefs) == 0 {
+				klog.Infof("Certificate %q has no owners and cannot be updated for ingress %q", tls.SecretName, ing.Name)
+				continue
+			}
+
+			notOwned := false
+			for _, ownerRef := range ownerRefs {
+				notOwned = notOwned || !reflect.DeepEqual(ingOwnerRef, ownerRef)
+			}
+
+			if notOwned {
+				klog.Infof("Certificate %q is not (solely) owned by ingress %q", tls.SecretName, ing.Name)
```

Comment:
Could you also update this log message to make it clear what, if any, action was taken as a result of this state?